controllers/cloudstackmachine_controller.go

```diff
@@ -70,93 +74,95 @@ func (r *CloudStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Re
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if capiMachine == nil {
 		log.Info("Waiting for CAPI cluster controller to set owner reference on CloudStack machine.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
 	// Fetch the CAPI Cluster.
-	cluster, retErr := util.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
-	if retErr != nil {
+	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, capiMachine.ObjectMeta)
+	if err != nil {
 		log.Info("Machine is missing cluster label or cluster does not exist.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
-	// Check the machine is not paused.
+	// Check if the machine is paused.
 	if annotations.IsPaused(cluster, csMachine) {
 		log.Info("CloudStackMachine or linked Cluster is paused. Requeuing reconcile.")
 		return reconcile.Result{}, nil
 	}
 
-	// Delete VM instance if deletion timestamp present.
-	if !csMachine.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(log, csMachine)
-	}
-
 	// Fetch the CloudStack cluster associated with this machine.
 	csCluster := &infrav1.CloudStackCluster{}
-	if retErr := r.Client.Get(
+	if err := r.Client.Get(
 		ctx,
 		client.ObjectKey{
 			Namespace: csMachine.Namespace,
 			Name:      cluster.Spec.InfrastructureRef.Name},
-		csCluster); retErr != nil {
-		if client.IgnoreNotFound(retErr) == nil {
+		csCluster); err != nil {
+		if client.IgnoreNotFound(err) == nil {
 			log.Info("CloudStackCluster not found.")
 			return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 		} else {
-			return ctrl.Result{}, retErr
+			return ctrl.Result{}, err
 		}
 	} else if csCluster.Status.ZoneID == "" {
 		log.Info("CloudStackCluster ZoneId not initialized. Likely not ready.")
 		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
 	}
 
+	// Delete VM instance if deletion timestamp present.
+	if !csMachine.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, log, csMachine, capiMachine, csCluster)
+	}
+
 	// Reconcile a VM instance for creates/updates
-	return r.reconcile(log, ctx, csCluster, csMachine, machine)
+	return r.reconcile(ctx, log, csMachine, capiMachine, csCluster)
 }
 
 // Actually reconcile/Create a VM instance.
 func (r *CloudStackMachineReconciler) reconcile(
-	log logr.Logger,
 	ctx context.Context,
-	csCluster *infrav1.CloudStackCluster,
+	log logr.Logger,
 	csMachine *infrav1.CloudStackMachine,
-	machine *capiv1.Machine) (ctrl.Result, error) {
+	capiMachine *capiv1.Machine,
+	csCluster *infrav1.CloudStackCluster) (ctrl.Result, error) {
 
 	log.V(1).Info("reconcile CloudStackMachine")
 
 	// Make sure bootstrap data is available in CAPI machine.
-	if machine.Spec.Bootstrap.DataSecretName == nil {
+	if capiMachine.Spec.Bootstrap.DataSecretName == nil {
 		log.Info("Bootstrap DataSecretName not yet available.")
 		return ctrl.Result{}, nil
 	}
-	log.Info("Got Bootstrap DataSecretName: " + *machine.Spec.Bootstrap.DataSecretName)
+	log.Info("Got Bootstrap DataSecretName.")
```

Comment:
why make this log less informative? Is it a security issue, or just spammy?

Reply:
It actually just prints what's already printed. So you end up seeing the name of the machine twice.

Apparently DataSecretName is just the name of the machine, which the logger is setup to print from the controller anyway.